en: add restart component of tidb cluster (#216)
pingcap/docs-tidb-operator: en/restart-a-tidb-cluster.md

# Restart a TiDB Cluster in Kubernetes

> If you find that the memory leak occurs in a Pod during use, you need to restart the cluster. This document describes how to gracefully rolling restart all the Pods in a component of the TiDB cluster, or gracefully logoff a Pod in the TiDB cluster and then restart the Pod using the gracefully restart command.

Reviewer: ```suggestion
If you find that the memory leak occurs in a Pod during use, you need to restart the cluster. This document describes how to gracefully perform a rolling restart to all the Pods in a component of the TiDB cluster, or gracefully log off a Pod in the TiDB cluster and then restart the Pod using the graceful restart command.
```

Author: How about `perform a graceful rolling restart`? I found in other sources that `graceful restart` is usually put together rather than split up.

Reviewer: Comment changed.

> **Warning:**
>
> It is not recommended to manually remove a Pod in the TiDB cluster without graceful restart in a production environment, because this might lead to some request failures of accessing the TiDB cluster though the `StatefulSet` controller pulls the Pod up again.

> ## Gracefully rolling restart all the Pods in a component

Reviewer: ```suggestion
## Gracefully performing a rolling restart to all Pods in a component
```

> 1. Refer to [Deploy TiDB on general Kubernetes](deploy-on-general-kubernetes.md) and modify the `${cluster_name}/tidb-cluster.yaml`.

Reviewer: ```suggestion
1. Refer to [Deploy TiDB on general Kubernetes](deploy-on-general-kubernetes.md) and modify the `${cluster_name}/tidb-cluster.yaml` file.
```

    In the following example, annotations of the `pd`, `tikv`, and `tidb` components are set, which means that all the Pods in these three components will be gracefully rolling restarted. You can set the annotation for a specific component according to your needs.

    ```yaml
    apiVersion: pingcap.com/v1alpha1
    kind: TidbCluster
    metadata:
      name: basic
    spec:
      version: v3.0.8
      timezone: UTC
      pvReclaimPolicy: Delete
      pd:
        baseImage: pingcap/pd
        replicas: 3
        requests:
          storage: "1Gi"
        config: {}
        annotations:
          tidb.pingcap.com/restartedAt: "202004201200"
      tikv:
        baseImage: pingcap/tikv
        replicas: 3
        requests:
          storage: "1Gi"
        config: {}
        annotations:
          tidb.pingcap.com/restartedAt: "202004201200"
      tidb:
        baseImage: pingcap/tidb
        replicas: 2
        service:
          type: ClusterIP
        config: {}
        annotations:
          tidb.pingcap.com/restartedAt: "202004201200"
    ```

2. Apply the update:

    {{< copyable "shell-regular" >}}

    ``` shell
    kubectl apply -f ${cluster_name} -n ${namespace}
    ```

> ## Gracefully restart a single Pod of the TiDB component

Reviewer: ```suggestion
## Gracefully perform a restart to a single Pod of the TiDB component
```

> This section describes how to gracefully restart a single Pod of the component in a TiDB cluster.

Reviewer: ```suggestion
This section describes how to gracefully perform a restart to a single Pod of the component in a TiDB cluster.
```
